src/osd/ReplicatedBackend.cc

```diff
@@ -2078,8 +2086,8 @@ int ReplicatedBackend::build_push_op(const ObjectRecoveryInfo &recovery_info,
        p != out_op->data_included.end();
        ++p) {
     bufferlist bit;
-    store->read(coll, recovery_info.soid,
-		     p.get_start(), p.get_len(), bit);
+    store->read(coll, recovery_info.soid, p.get_start(), p.get_len(), bit,
+                cache ? CEPH_OSD_OP_FLAG_FADVISE_DONTNEED: 0);
```

Comment:
To me, it should be cache ? 0 : CEPH_OSD_OP_FLAG_FADVISE_DONTNEED.  That is, cache = true should mean that we are going to cache the data read.  I'd actually prefer that it had a more descriptive name like ignore_cache or cache_dont_need.